user_sync/rules.py

```diff
@@ -68,27 +65,21 @@ def __init__(self, caller_options):
         self.organization_info_by_organization = {}
         self.adding_dashboard_user_key = set()
 
-        '''
-        remove_user_key_list = options['remove_user_key_list']
-        remove_user_key_list = set(remove_user_key_list) if (remove_user_key_list != None) else set()
-        self.remove_user_key_list = remove_user_key_list
-        
-        delete_user_key_list = options['delete_user_key_list']
-        delete_user_key_list = set(delete_user_key_list) if (delete_user_key_list != None) else set()
-        self.delete_user_key_list = delete_user_key_list
-        '''
-        
         # we only need a reference to either the remove or delete key list,
         # since you can't do both
-        remove_or_delete_user_key_list = options['remove_user_key_list'] if options['remove_user_key_list'] != None else options['delete_user_key_list']
-        remove_or_delete_user_key_list = set(remove_or_delete_user_key_list) if (remove_or_delete_user_key_list != None) else set()
-        self.remove_or_delete_user_key_list = remove_or_delete_user_key_list
+        self.orphan_user_key_list = set(options['remove_user_key_list'] or options['delete_user_key_list'] or [])
+        self.orphan_list_output_path = options['remove_list_output_path'] or options['delete_list_output_path']
+        
+        # determine if we need to delete user accounts instead of just removing
+        # them
+        self.delete_user_accounts = options['delete_nonexistent_users'] or options['delete_user_key_list']
+        
+        # determine how we should treat orphaned accounts during processing
+        self.process_orphaned_accounts_for_delete = options['delete_list_output_path'] is not None or options['delete_nonexistent_users']
```

Comment:
take out "is not None", it's not necessary (and is overly strict, even if delete_list_output_path was specified as an empty string we want the test to fail).  Thus = "options[...] or options[...]" not "options[...] is not None or options[...]".